src/utils.ts

```diff
@@ -1004,6 +1004,14 @@ export function makeInterruptibleAsyncInterval(
     // never completes, the `timeUntilNextCall` will continue to grow
     // negatively unbounded, so it will never trigger a reschedule here.
 
+    // This is possible in virtualized environments like AWS Lambda where our
+    // clock is unreliable. In these cases the timer is "running" but never
+    // actually completes, so we want to execute immediately and then attempt
+    // to reschedule.
+    if (timeUntilNextCall < 0) {
+      executeAndReschedule();
```

Comment:
I think you probably want to add a `return` after `executeAndReschedule` if you move it here, since it otherwise used to be the last statement in the function. Otherwise, I think it looks okay. If you wanted to double check I think I made the unit tests for this class pass in a custom clock you can mess around with.

Reply:
Added: https://github.com/mongodb/node-mongodb-native/pull/2801/commits/c0abff66087722859560a300f203056efe9a0805

Reply:
LGTM. Do you think its worth adding a test to cover this, since not adding a `return` statement caused no failure? I only suggest it because this code is pretty critical, and often hard to debug when something is wrong (i.e. the Lambda unreliable time situation)

Reply:
@durran ping

Reply:
I'll add a test for that.